<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../../img/favicon.ico">
  <title>Datsets - Kipoiseq</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Datsets";
    var mkdocs_page_input_path = "datasets/sequence.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../../js/jquery-2.1.1.min.js" defer></script>
  <script src="../../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href="../.." class="icon icon-home"> Kipoiseq</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="../..">Home</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">Datsets</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#beddataset">BedDataset</a></li>
    

    <li class="toctree-l2"><a href="#seqstringdataset">SeqStringDataset</a></li>
    

    <li class="toctree-l2"><a href="#seqdataset">SeqDataset</a></li>
    

    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Transforms</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../../transforms/functional/">Functional</a>
                </li>
                <li class="">
                    
    <a class="" href="../../transforms/transforms/">Class-based</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../../extractors/">Extractors</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../..">Kipoiseq</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../..">Docs</a> &raquo;</li>
    
      
    
    <li>Datsets</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="http://github.com/kipoi/kipoiseq/edit/master/docs/datasets/sequence.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h2 id="beddataset">BedDataset</h2>
<pre><code class="python">BedDataset(self, tsv_file, label_dtype=None, bed_columns=3, num_chr=False, ambiguous_mask=None, incl_chromosomes=None, excl_chromosomes=None)
</code></pre>

<p>Reads a tsv file in the following format:</p>
<pre><code>chr  start  stop  task1  task2 ...
</code></pre>

<p><strong>Arguments</strong></p>
<ul>
<li><strong>tsv_file</strong>: tsv file type</li>
<li><strong>bed_columns</strong>: number of columns corresponding to the bed file. All the columns
    after that will be parsed as targets</li>
<li><strong>num_chr</strong>: if specified, 'chr' in the chromosome name will be dropped</li>
<li><strong>label_dtype</strong>: specific data type for labels</li>
<li><strong>ambiguous_mask</strong>: if specified, rows containing only ambiguous_mask values will be skipped</li>
<li><strong>incl_chromosomes</strong>: exclusive list of chromosome names to include in the final dataset.
    if not None, only these will be present in the dataset</li>
<li><strong>excl_chromosomes</strong>: list of chromosome names to omit from the dataset.</li>
</ul>
<h2 id="seqstringdataset">SeqStringDataset</h2>
<pre><code class="python">SeqStringDataset(self, intervals_file, fasta_file, num_chr_fasta=False, label_dtype=None, auto_resize_len=None, use_strand=False, force_upper=True)
</code></pre>

<p>Dataloader for a combination of fasta and tab-delimited input files such as bed files. The dataloader extracts regions from the fasta file as defined in the tab-delimited <code>intervals_file</code>. Returned sequences are of the type np.array([str]).</p>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>intervals_file</strong>: bed3+<columns> file path containing intervals + (optionally) labels <a href="https://raw.githubusercontent.com/kipoi/kipoiseq/kipoi_dataloader/tests/data/sample_intervals.bed">example</a></li>
<li><strong>fasta_file</strong>: Reference genome FASTA file path. <a href="https://raw.githubusercontent.com/kipoi/kipoiseq/kipoi_dataloader/tests/data/sample.5kb.fa">example</a></li>
<li><strong>num_chr_fasta</strong>: True, the the dataloader will make sure that the chromosomes don't start with chr.</li>
<li><strong>label_dtype</strong>: None, datatype of the task labels taken from the intervals_file. Allowed - string', 'int', 'float', 'bool'</li>
<li><strong>auto_resize_len</strong>: None, required sequence length.</li>
<li><strong>use_strand</strong>: reverse-complement fasta sequence if bed file defines negative strand</li>
<li>
<p><strong>force_upper</strong>: Force uppercase output of sequences
<strong>Output schema</strong></p>
</li>
<li>
<p><strong><em>inputs</em></strong>: </p>
<ul>
<li><strong>shape=()</strong>, DNA sequence as string</li>
</ul>
</li>
<li><strong><em>targets</em></strong>: <ul>
<li><strong>shape=(None,)</strong>, (optional) values following the bed-entry - chr  start  end  target1   target2 ....</li>
</ul>
</li>
<li><strong><em>metadata</em></strong>: <ul>
<li><strong><em>ranges</em></strong>: <ul>
<li>Genomic ranges: chr, start, end, name, strand</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="seqdataset">SeqDataset</h2>
<pre><code class="python">SeqDataset(self, intervals_file, fasta_file, num_chr_fasta=False, label_dtype=None, auto_resize_len=None, use_strand=False, alphabet_axis=1, dummy_axis=None, alphabet='ACGT')
</code></pre>

<p>Dataloader for a combination of fasta and tab-delimited input files such as bed files. The dataloader extracts regions from the fasta file as defined in the tab-delimited <code>intervals_file</code> and converts them into one-hot encoded format. Returned sequences are of the type np.array with the shape inferred from the arguments: <code>alphabet_axis</code> and <code>dummy_axis</code>.</p>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>intervals_file</strong>: bed3+<columns> file path containing intervals + (optionally) labels <a href="https://raw.githubusercontent.com/kipoi/kipoiseq/kipoi_dataloader/tests/data/sample_intervals.bed">example</a></li>
<li><strong>fasta_file</strong>: Reference genome FASTA file path. <a href="https://raw.githubusercontent.com/kipoi/kipoiseq/kipoi_dataloader/tests/data/sample.5kb.fa">example</a></li>
<li><strong>num_chr_fasta</strong>: True, the the dataloader will make sure that the chromosomes don't start with chr.</li>
<li><strong>label_dtype</strong>: None, datatype of the task labels taken from the intervals_file. Allowed - string', 'int', 'float', 'bool'</li>
<li><strong>auto_resize_len</strong>: None, required sequence length.</li>
<li><strong>use_strand</strong>: reverse-complement fasta sequence if bed file defines negative strand</li>
<li><strong>alphabet_axis</strong>: axis along which the alphabet runs (e.g. A,C,G,T for DNA)</li>
<li><strong>dummy_axis</strong>: defines in which dimension a dummy axis should be added. None if no dummy axis is required.</li>
<li><strong>alphabet</strong>: alphabet to use for the one-hot encoding. This defines the order of the one-hot encoding. Can either be a list or a string: 'DNA', 'RNA', 'AMINO_ACIDS'.</li>
</ul>
<p><strong>Output schema</strong></p>
<ul>
<li><strong><em>inputs</em></strong>: <ul>
<li><strong>shape=(None, 4)</strong>, One-hot encoded DNA sequence</li>
</ul>
</li>
<li><strong><em>targets</em></strong>: <ul>
<li><strong>shape=(None,)</strong>, (optional) values following the bed-entry - chr  start  end  target1   target2 ....</li>
</ul>
</li>
<li><strong><em>metadata</em></strong>: <ul>
<li><strong><em>ranges</em></strong>: <ul>
<li>Genomic ranges: chr, start, end, name, strand</li>
</ul>
</li>
</ul>
</li>
</ul>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../../transforms/functional/" class="btn btn-neutral float-right" title="Functional">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../.." class="btn btn-neutral" title="Home"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="http://github.com/kipoi/kipoiseq/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href="../.." style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../../transforms/functional/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '../..';</script>
    <script src="../../js/theme.js" defer></script>
      <script src="../../search/main.js" defer></script>

</body>
</html>
